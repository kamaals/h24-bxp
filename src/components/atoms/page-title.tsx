import React from "react";

function PageTitle({ title }: { title: string }) {
  return (
    <h1 className={"text-2xl md:text-3xl lg:text-4xl font-bold font-sans"}>
      {title}
    </h1>
  );
}

export default PageTitle;
