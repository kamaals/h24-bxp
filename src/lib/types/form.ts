/* istanbul ignore file @preserve */
import {
  useCreateCategoryMutation,
  useUpdateCategoryMutation,
} from "@/lib/store/api/categoryServices";
import {
  useUpdateProductMutation,
  useCreateProductMutation,
} from "@/lib/store/api/productServices";

export type GenericMutationType =
  | typeof useCreateCategoryMutation
  | typeof useUpdateCategoryMutation
  | typeof useCreateProductMutation
  | typeof useUpdateProductMutation;
