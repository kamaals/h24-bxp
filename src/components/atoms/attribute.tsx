import React from "react";
import { AttributeType, AttributeTypes } from "@/lib/types/product";
import { cn } from "@/lib/utils";

function Attribute({ attribute }: { attribute: AttributeType }) {
  const type = attribute.type as AttributeTypes;
  const bgClassName = () => {
    switch (type) {
      case "boolean":
        return "bg-rose-100";
      case "number":
        return "bg-violet-100";
      case "tag":
        return "bg-lime-100";
      case "text":
        return "bg-emerald-100";
      case "url":
        return "bg-cyan-100";
      default:
        return "bg-white";
    }
  };

  return (
    <span className={cn("rounded-lg text-xs p-2 inline-block", bgClassName())}>
      <span className={"flex gap-1"}>
        <span className="font-bold">{attribute.code}: </span>
        <span>{attribute.name}</span>
      </span>
      <span>{attribute.type}</span>
    </span>
  );
}

export default Attribute;
