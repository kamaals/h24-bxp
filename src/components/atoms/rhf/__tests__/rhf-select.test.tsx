import { describe, expect, it, beforeAll } from "vitest";
import { render, screen, cleanup } from "@testing-library/react";
import RHFSelect from "../rhf-select";
import { useForm } from "react-hook-form";
import { Form } from "@/components/atoms/form";
import React from "react";
import { SelectItem } from "@/components/atoms/select";

const FormProvider = ({
  children,
  value,
  error,
}: {
  children: React.ReactNode;
  value?: string;
  error?: string;
}) => {
  const form = useForm<{ name: string }>({
    defaultValues: { name: value },
  });

  React.useEffect(() => {
    if (error) {
      form.setError("name", { message: error });
    }
  }, [error, form]);

  return <Form {...form}>{children}</Form>;
};

describe(RHFSelect.name, () => {
  beforeAll(() => {
    cleanup();
  });

  it("should match the snapshot", () => {
    const result = render(
      <FormProvider value={"Alpha"}>
        <RHFSelect name={"name"} label={"Name"}>
          <SelectItem value="text">Text</SelectItem>
          <SelectItem value="tag">Tag</SelectItem>
        </RHFSelect>
      </FormProvider>,
    );
    expect(result).toMatchSnapshot();
  });

  it("should render", () => {
    const { container } = render(
      <FormProvider>
        <RHFSelect name={"name"} label={"Name"}>
          <SelectItem value="text">Text</SelectItem>
          <SelectItem value="tag">Tag</SelectItem>
        </RHFSelect>
      </FormProvider>,
    );
    const nameInput = container.querySelector(
      'button[data-slot="select-trigger"]',
    ) as HTMLInputElement;
    expect(nameInput).toBeInTheDocument();
  });

  it("should trigger error", () => {
    render(
      <FormProvider error={"Not valid"}>
        <RHFSelect name={"name"} label={"Name"}>
          <SelectItem value="text">Text</SelectItem>
          <SelectItem value="tag">Tag</SelectItem>
        </RHFSelect>
      </FormProvider>,
    );
    const errorText = screen.getByText("Not valid");
    expect(errorText).toBeInTheDocument();
  });
});
