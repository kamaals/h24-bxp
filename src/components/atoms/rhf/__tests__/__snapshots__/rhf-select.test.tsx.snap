// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RHFSelect > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid gap-2"
        data-slot="form-item"
      >
        <div>
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
            data-error="false"
            data-slot="form-label"
            for="«r0»-form-item"
          >
            Name
          </label>
          <p
            class="text-muted-foreground text-xs"
            data-slot="form-description"
            id="«r0»-form-item-description"
          />
        </div>
        <div
          aria-describedby="«r0»-form-item-description"
          aria-invalid="false"
          class="space-y-1 mt-0.5"
          data-slot="form-control"
          id="«r0»-form-item"
        >
          <button
            aria-autocomplete="none"
            aria-controls="radix-«r1»"
            aria-expanded="false"
            class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-[180px]"
            data-size="default"
            data-slot="select-trigger"
            data-state="closed"
            dir="ltr"
            role="combobox"
            type="button"
          >
            <span
              data-slot="select-value"
              style="pointer-events: none;"
            />
            <svg
              aria-hidden="true"
              class="lucide lucide-chevron-down size-4 opacity-50"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m6 9 6 6 6-6"
              />
            </svg>
          </button>
          <span
            class="inline-block h-2 bg-emerald-300 w-2 ml-1 rounded-full"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid gap-2"
      data-slot="form-item"
    >
      <div>
        <label
          class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 data-[error=true]:text-destructive"
          data-error="false"
          data-slot="form-label"
          for="«r0»-form-item"
        >
          Name
        </label>
        <p
          class="text-muted-foreground text-xs"
          data-slot="form-description"
          id="«r0»-form-item-description"
        />
      </div>
      <div
        aria-describedby="«r0»-form-item-description"
        aria-invalid="false"
        class="space-y-1 mt-0.5"
        data-slot="form-control"
        id="«r0»-form-item"
      >
        <button
          aria-autocomplete="none"
          aria-controls="radix-«r1»"
          aria-expanded="false"
          class="border-input data-[placeholder]:text-muted-foreground [&_svg:not([class*='text-'])]:text-muted-foreground focus-visible:border-ring focus-visible:ring-ring/50 aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive dark:bg-input/30 dark:hover:bg-input/50 flex items-center justify-between gap-2 rounded-md border bg-transparent px-3 py-2 text-sm whitespace-nowrap shadow-xs transition-[color,box-shadow] outline-none focus-visible:ring-[3px] disabled:cursor-not-allowed disabled:opacity-50 data-[size=default]:h-9 data-[size=sm]:h-8 *:data-[slot=select-value]:line-clamp-1 *:data-[slot=select-value]:flex *:data-[slot=select-value]:items-center *:data-[slot=select-value]:gap-2 [&_svg]:pointer-events-none [&_svg]:shrink-0 [&_svg:not([class*='size-'])]:size-4 w-[180px]"
          data-size="default"
          data-slot="select-trigger"
          data-state="closed"
          dir="ltr"
          role="combobox"
          type="button"
        >
          <span
            data-slot="select-value"
            style="pointer-events: none;"
          />
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down size-4 opacity-50"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
        <span
          class="inline-block h-2 bg-emerald-300 w-2 ml-1 rounded-full"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
